src/core/services/NetworkCommunicatorService.ts

```diff
@@ -0,0 +1,103 @@
+import axios from "axios";
+import { Server } from "node:http";
+import { EventEmitter } from "node:events";
+import { sign, verify, JsonWebTokenError } from "jsonwebtoken";
+import express, { Express } from "express";
+import config from "@config";
+import { log as mainLogger } from "@core/logging";
+import { wrapAsync } from "@core/utils";
+import { extractToken } from "@core/utils/auth";
+
+const log = mainLogger.child({ app: "network-communicator-service" });
+
+class NetworkCommunicatorService {
+  private server?: Server;
+  private events = new EventEmitter();
+
+  /**
+   * Sign a internal service request
+   * @param payload
+   * @returns
+   */
+  private sign(payload: string | Buffer | object = {}) {
+    return sign(payload, config.secret, { algorithm: "HS256" });
+  }
+
+  /**
+   * Verify a internal service request
+   * @param token
+   * @returns the payload decoded if the signature is valid and optional expiration, audience, or issuer are valid. If not valid, it will return the JsonWebTokenError.
+   */
+  private verify(authHeader?: string) {
+    const token = extractToken(authHeader);
+    if (!token) {
+      return new JsonWebTokenError("No token found");
+    }
+    try {
+      return verify(token, config.secret);
```

Comment:
I think it would be slightly more secure to define a new secret rather than reusing this one, because the latter is only known within beabee core (API, legacy app, webhook app) at the moment, and technically the Telegram bot is an "external" integration (i.e. only has access via the API)

Reply:
Okay I will define a new one

Reply:
I'ts called `BEABEE_SERVICE_SECRET` now